diff --git a/lib/dolphin/os/OSAlloc.cpp b/lib/dolphin/os/OSAlloc.cpp
--- a/lib/dolphin/os/OSAlloc.cpp
+++ b/lib/dolphin/os/OSAlloc.cpp
@@ -9,7 +9,7 @@
   return CaptureStackBackTrace(0, max_frames, buffer, NULL);
 }
 #define backtrace capture_backtrace
-#elif defined(__ANDROID__)
+#elif defined(__ANDROID__) || defined(__SWITCH__)
 static inline int capture_backtrace(void** buffer, int max_frames) {
   (void)buffer;
   (void)max_frames;
